serde_derive/src/ser.rs

```diff
@@ -723,6 +723,7 @@ fn serialize_adjacently_tagged_variant(
             where
                 __S: _serde::Serializer,
             {
+                #[allow(unused_variables)]
```

Comment:
Please add a comment as a reminder of why this needs to be here, i.e. under what circumstances the variables would be unused.